zap-widget/src/components/SearchBox.tsx

```diff
@@ -0,0 +1,172 @@
+function SearchBox() {
+  const [packageQueryName, setPackageQueryName] = useState('')
+  const [result, setResult] = useState<ResultType | null>(null)
+  const [isLoading, setIsLoading] = useState(false)
+  const [amountInSats, setAmountInSats] = useState(0)
+
+  const { invoice, setInvoice } = useClient()
+
+  useEffect(() => {
+    const decodeInv = () => {
+      if (invoice.length > 1) {
+        const amountInSat = decode.decode(invoice) as { sections: { value?: string | number }[] }
+        setAmountInSats(+amountInSat.sections[2]?.value / 1000)
+      }
+    }
+
+    decodeInv()
+  }, [invoice])
+
+  const fetchPackage = async () => {
+    try {
+      //clear the result and input alone - when the cncel icon is clicked
+      if (result) {
+        setResult(null)
+        setPackageQueryName('')
+        setInvoice('')
+        setAmountInSats(0)
+        return
+      }
+
+      //fetch the package
+      setIsLoading(true)
+      if (packageQueryName.length > 1) {
+        const response = await fetch(`https://registry.npmjs.org/${packageQueryName}/latest`)
+        const packageInfo = await response.json()
+        console.log(packageInfo)
+
+        if (!packageInfo || packageInfo === 'Not Found') {
+          setResult({
+            warn: true,
+            packageName: packageQueryName,
+            hint: '⚠️ Can’t find this package. Typo or doesn’t exist? 🤔',
+          })
+          return
+        } else if (!packageInfo.funding || packageInfo.funding.type !== 'lightning') {
+          setResult({
+            warn: true,
+            packageName: packageQueryName,
+            hint: '⚠️ This package does not accept bitcoin tips yet 🥺',
+          })
+          return
+        } else {
+          const lnAddress = packageInfo.funding.url
+
+          setResult({
+            warn: false,
+            lnAddress: lnAddress.startsWith('lightning:') ? lnAddress.substring(10) : lnAddress,
+            packageName: packageInfo.name,
+            description: packageInfo.description,
+          })
+        }
+      }
+    } catch (err) {
+      console.error('Error fetching package:', err)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen mx-8">
+      <div className="w-full flex items-center justify-center flex-col md:flex-row gap-3">
+        <input
+          value={packageQueryName}
+          onChange={(e) => setPackageQueryName(e.target.value)}
+          placeholder="Search package name..."
+          className="w-full md:w-[576px] h-12 md:h-[72px] rounded-full pl-5 py-2 bg-zap-gradient border border-white/25 text-2xl font-grotesk placeholder-opacity-70"
+        />
+        <button
+          onClick={fetchPackage}
+          disabled={isLoading}
+          className={`hover:invert w-full md:w-[120px] h-12 md:h-[72px] flex items-center justify-center gap-2 rounded-full font-bold text-2xl md:text-3xl transition-all duration-200 ${
```

Comment:
Actually for large screens, you should not specify the width and instead just use `p-6`, since hardcoding widths is always risky (for the search bar it is fine because it is a text input, but here the text is fixed), so for example in figma:

<img width="262" height="253" alt="Screenshot 2025-07-22 at 4 43 18 PM" src="https://github.com/user-attachments/assets/5665d5bc-112c-4064-bd42-df4c01b72e04" />

But because this would break the width when replaced with "X" icon, we can proceed with `w-24`

<img width="370" height="260" alt="Screenshot 2025-07-22 at 4 46 23 PM" src="https://github.com/user-attachments/assets/94d92f56-f76d-476b-84fb-4211f4bbbf9b" />